### PR TITLE
Remove ubi8 workflows, add job to build udi9

### PR DESCRIPTION
**Pull request by dkwon17**
Part of https://github.com/eclipse-che/che/issues/23034

This PR
* updates the README
* updates the contributing guide
* removes the ubi8 and udi8 build jobs
* adds the udi9 build job

Currently we have:
```
https://quay.io/repository/devfile/base-developer-image:ubi9-latest

https://quay.io/repository/devfile/base-developer-image:ubi8-latest     ------
                                                                              |-- same image
https://quay.io/repository/devfile/base-developer-image:latest     -----------
```

After this PR, the `ubi9-latest` and `latest` tags will refer to the same image for both `base-developer-image` and `universal-developer-image`:
```
https://quay.io/repository/devfile/base-developer-image:ubi8-latest

https://quay.io/repository/devfile/base-developer-image:ubi9-latest     ------
                                                                              |-- same image
https://quay.io/repository/devfile/base-developer-image:latest     -----------
```

### CODE REVIEW

File: .github/workflows/ubi9-build.yaml [hunk 1]
```diff
@@ -47,3 +60,54 @@ jobs:
...
+          sudo rm -rf /usr/local/lib/android
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            quay.io/devfile/universal-developer-image
+          flavor: |
+            latest=true
+            prefix=ubi8-,onlatest=true
```
Comment (svor) [inline .github/workflows/ubi9-build.yaml:84]: should be ubi9

File: .github/workflows/ubi9-build.yaml [hunk 2]
```diff
@@ -47,3 +60,54 @@ jobs:
...
+          tags: |
+            type=sha,prefix=ubi8-,format=short
```
Comment (svor) [inline .github/workflows/ubi9-build.yaml:86]: should be ubi9

File: .github/workflows/ubi9-build.yaml (hunk 1 above)
Comment (dkwon17) [reply .github/workflows/ubi9-build.yaml]: Thank you, fixed

File: .github/workflows/ubi9-build.yaml (hunk 2 above)
Comment (dkwon17) [reply .github/workflows/ubi9-build.yaml]: Thank you, fixed

**Review by ibuziuk — APPROVED** [review]
Great Job!

**Review by tolusha — APPROVED** [review]

**Pull request merged**